src/cborencoder.c

```diff
@@ -117,16 +117,16 @@
  *      CborEncoder encoder, mapEncoder;
  *      cbor_encoder_init(&encoder, buf, sizeof(buf), 0);
  *      err = cbor_encoder_create_map(&encoder, &mapEncoder, 1);
- *      if (!err)
+ *      if (err)
```

Comment:
This will work.  However, the return type is not actually a boolean its an enum.  Shouldn't we be using the enum values?

(i.e. `if (err != CborNoError) return err;` )

Reply:
`CborNoError` is specifically the value 0 so this kind of construct is possible. But if someone contributes a cleanup to compare to `CborNoError`, I'll probably accept.